### PR TITLE
layers: Add Add() and Destroy() for handle map management

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
These methods must be used to insert and remove state into the handle -> state object maps. Having this all happen the same way will be important for upcoming locking changes.

### CODE REVIEW

File: layers/gpu_utils.h [hunk 1]
```diff
@@ -390,9 +390,9 @@ void UtilSubmitBarrier(VkQueue queue, ObjectType *object_ptr) {
 
         uint32_t queue_family_index = 0;
 
-        auto queue_state_it = object_ptr->queueMap.find(queue);
-        if (queue_state_it != object_ptr->queueMap.end()) {
-            queue_family_index = queue_state_it->second->queueFamilyIndex;
+        auto queue_state = object_ptr->GetQueueState(queue);
```
Comment (ncesario-lunarg) [inline layers/gpu_utils.h:393]: `Get<QUEUE_STATE>`?

File: layers/state_tracker.h [hunk 2]
```diff
@@ -267,37 +267,63 @@ class ValidationStateTracker : public ValidationObject {
...
-    VALSTATETRACK_MAP_AND_TRAITS(VkSampler, SAMPLER_STATE, samplerMap)
-    VALSTATETRACK_MAP_AND_TRAITS(VkImageView, IMAGE_VIEW_STATE, imageViewMap)
-    VALSTATETRACK_MAP_AND_TRAITS(VkImage, IMAGE_STATE, imageMap)
-    VALSTATETRACK_MAP_AND_TRAITS(VkBufferView, BUFFER_VIEW_STATE, bufferViewMap)
-    VALSTATETRACK_MAP_AND_TRAITS(VkBuffer, BUFFER_STATE, bufferMap)
-    VALSTATETRACK_MAP_AND_TRAITS(VkPipeline, PIPELINE_STATE, pipelineMap)
-    VALSTATETRACK_MAP_AND_TRAITS(VkDeviceMemory, DEVICE_MEMORY_STATE, memObjMap)
-    VALSTATETRACK_MAP_AND_TRAITS(VkFramebuffer, FRAMEBUFFER_STATE, frameBufferMap)
-    VALSTATETRACK_MAP_AND_TRAITS(VkShaderModule, SHADER_MODULE_STATE, shaderModuleMap)
+    VALSTATETRACK_MAP_AND_TRAITS(VkQueue, QUEUE_STATE, queue_map_)
```
Comment (ncesario-lunarg) [inline layers/state_tracker.h:270]: :tada: more style normalization :)

**Review by ncesario-lunarg — APPROVED** [review]
Nice!

**Review by ziga-lunarg — APPROVED** [review]

**Review by Tony-LunarG — APPROVED** [review]
LGTM

**Pull request merged**